client/components/goals.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import AppContext from '../lib/app-context';
+import GoalsForm from './goals-form';
+import Goal from './individual-goal';
+
+export default class Goals extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      goals: []
+    };
+    this.addGoal = this.addGoal.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    fetch(`/api/goals/${this.context.user.userId}`)
+      .then(res => res.json())
+      .then(goals => this.setState({
+        goals
+      }))
+      .catch(err => console.error(err));
+  }
+
+  addGoal(newGoal) {
+    const req = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newGoal)
+    };
+    fetch(`/api/goals/${this.context.user.userId}`, req)
+      .then(res => res.json())
+      .then(goal => {
+        this.state.goals.push(goal);
```

Comment:
We try to always use `.concat` or make a copy with `.slice()` before modifying arrays in state. state updates are all about "replacing" stuff. so in the case of arrays, it's like making a new egg carton, transferring the eggs out of their old carton, then adding the new egg to the new carton.